Update harness-variables.md wrt <+env.envGroup..> expressions (#1906)
harness/developer-hub: docs/platform/12_Variables-and-Expressions/harness-variables.md

### <+env.envGroupName>

> The name of the environment group to which the environment belongs (if defined). This only resolves if the deployment was done to an Environment Group

Reviewer: ```suggestion
The name of the environment group to which the environment belongs (if defined). This only resolves if the deployment was done to an environment group
```

> The name of the environment group to which the environment belongs (if defined). This expression resolves only if the deployment is done to an environment group

Reviewer: ```suggestion
The name of the environment group to which the environment belongs (if defined). This expression resolves only if the deployment is done to an environment group.
```

### <+env.envGroupRef>

> The environment group reference. This only resolves if the deployment was done to an Environment Group

Reviewer: ```suggestion
The environment group reference. This expression resolves only if the deployment is done to an environment group
```

> The environment group reference. This expression resolves only if the deployment is done to an environment group

Reviewer: ```suggestion
The environment group reference. This expression resolves only if the deployment is done to an environment group.
```
